lib/rubocop/cop/style/multiline_when_then.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks uses of the `then` keyword
+      # in multi-line when statements.
+      #
+      # @example
+      #   # bad
+      #   case foo
+      #   when bar then
+      #   end
+
+      #   # good
+      #   case foo
+      #   when bar
+      #   end
+
+      #   # good
+      #   case foo
+      #   when bar then do_something
+      #   end
```

Comment:
It needs to be connected with `#`.

```diff
      # @example
      #   # bad
      #   case foo
      #   when bar then
      #   end
-
+     #
      #   # good
      #   case foo
      #   when bar
      #   end
-
+     #
      #   # good
      #   case foo
      #   when bar then do_something
      #   end
```

Some of example code generated for document is lost.
https://github.com/rubocop-hq/rubocop/pull/7114/files#diff-3f2ebe9c9435b29358487264049625a4